main.ts

```diff
@@ -97,11 +102,42 @@ class Translator {
       this.emit('>');
     }
     this.visitEachIfPresent(decl.heritageClauses);
+    // Check for @IMPLEMENTS interfaces to add.
+    // TODO(martinprobst): Drop all special cases for @SOMETHING after migration to TypeScript.
+    var implIfs = this.getImplementsDecorators(decl.decorators);
+    if (implIfs.length > 0) {
+      // Check if we have to emit an 'implements ' or a ', '
+      if (decl.heritageClauses && decl.heritageClauses.length > 0 &&
+          decl.heritageClauses.some((hc) => hc.token === ts.SyntaxKind.ImplementsKeyword)) {
+        // There was some implements clause.
+        this.emit(',');
+      } else {
+        this.emit('implements');
+      }
+      this.emit(implIfs.join(' , '));
+    }
     this.emit('{');
     this.visitEachIfPresent(decl.members);
     this.emit('}');
   }
 
+  /** Returns parameters to @IMPLEMENTS, returns their identifiers. */
+  getImplementsDecorators(decorators: ts.NodeArray<ts.Decorator>): string[] {
+    var interfaces = [];
+    if (!decorators) return interfaces;
```

Comment:
That's odd indeed. I checked the spec:
https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.10.3

This sounds like it's a bug, at least I cannot find language in the spec to the point of converting `Array<any>` to `Array<string>`.